src/rustdoc_gen.rs

```diff
@@ -48,13 +48,17 @@ impl<'a> CrateSource<'a> {
                 .filter_map(|dep| dep.is_optional().then_some(dep.name()))
                 .map(|x| x.to_string())
                 .collect(),
-            Self::ManifestPath { manifest } => manifest
-                .parsed
-                .dependencies
-                .iter()
-                .filter_map(|(name, dep)| dep.optional().then_some(name))
-                .map(|x| x.to_string())
-                .collect(),
+            Self::ManifestPath { manifest } => {
+                let mut dependencies = manifest.parsed.dependencies.clone();
+                for target in manifest.parsed.target.values() {
+                    dependencies.extend(target.dependencies.clone().into_iter());
+                }
```

Comment:
How does enabling target-specific dependencies behave when the target isn't being built?

Reply:
```
cargo new targetspecific
cd targetspecific
echo "[target.'cfg(windows)'.dependencies]" >> Cargo.toml
echo 'async-std = { version = "1.6.2", optional = true }' >> Cargo.toml
cargo run --features async-std
```
It seems to work on a unix machine.

Reply:
Seems like that would be a good test to have in the repo documented specifically for this purpose rather than relying only on the third-party reproductions

Reply:
Done in 2c2e7c991f32a8e8ac514e00577787a2adb47d40.